staging/src/k8s.io/apimachinery/pkg/version/types.go

```diff
@@ -20,15 +20,25 @@ package version
 // TODO: Add []string of api versions supported? It's still unclear
 // how we'll want to distribute that information.
 type Info struct {
-	Major        string `json:"major"`
-	Minor        string `json:"minor"`
-	GitVersion   string `json:"gitVersion"`
-	GitCommit    string `json:"gitCommit"`
-	GitTreeState string `json:"gitTreeState"`
-	BuildDate    string `json:"buildDate"`
-	GoVersion    string `json:"goVersion"`
-	Compiler     string `json:"compiler"`
-	Platform     string `json:"platform"`
+	// Major is the major version of the effective version
+	Major string `json:"major"`
+	// Minor is the minor version of the effective version
+	Minor string `json:"minor"`
```

Comment:
trying to understand the implications here.
so if one passes --version and  the new compatibility-version flags on the cli they will get varying output?
if so, this pr touches not only the /version endpoints.